packages/patternfly-4/react-table/src/components/Table/utils/utils.tsx

```diff
@@ -33,3 +33,73 @@ export const isRowExpanded = (row: IRow, rows: IRow[]) => {
   }
   return undefined;
 };
+
+export const getErrorTextByValidator = (validatorName: string, validators: IValidatorDef[]) => {
+  const result = validators.filter(validator => {
+    return validator.name === validatorName;
+  });
+  return result[0].errorText;
+};
+
+export const cancelCellEdits = (row: IRow) => {
+  (row.cells as IRowCell[]).forEach(cell => {
+    delete cell.props.errorText;
+    delete cell.props.editableValue;
+    cell.props.isValid = true;
+  });
+
+  row.isEditable = !row.isEditable;
+  row.isValid = true;
+  return row;
+};
+
+export const validateCellEdits = (row: IRow, type: RowEditType, validationErrors: RowErrors) => {
+  row.isValid = !!Object.keys(validationErrors).length ? false : true;
```

Comment:
Does the `!!` do something specific here? It seems like it can be removed.

Reply:
Maybe we can use `!Boolean(Object.keys(...).length)`